src/TACSAssembler.cpp

```diff
@@ -4171,6 +4173,15 @@ void TACSAssembler::assembleRes(TACSBVec *residual) {
       naux = auxElements->getAuxElements(&aux);
     }
 
+    // To avoid allocating memory inside the element loop, make the aux element
+    // contribution array big enough for the largest element
+    int maxNVar = this->maxElementSize;
+    TacsScalar *auxElemRes;
+    bool scaleAux = lambda != TacsScalar(1.0) && naux > 0;
```

Comment:
Assign using an if condition? We don't use this type of construct elsewhere in TACS.

Reply:
I do this because this condition is checked in multiple places in the function, and it avoids having to check the same condition repeatedly inside the loop over all elements. Do you have a preffered way of doing this?

Reply:
I think he means something like this:

```
bool scaleAux = False; 
if (lambda != TacsScalar(1.0) && naux > 0){
  scaleAux = True;
}
```